packages/browser-wallet/src/popup/pages/Web3ProofRequest/Web3ProofRequest.tsx

```diff
@@ -0,0 +1,186 @@
+import { addToastAtom } from '@popup/state';
+import { useDecryptedSeedPhrase } from '@popup/shared/utils/seed-phrase-helpers';
+import { getGlobal, getNet } from '@shared/utils/network-helpers';
+import { BackgroundResponseStatus, ProofBackgroundResponse } from '@shared/utils/types';
+import PendingArrows from '@assets/svg/pending-arrows.svg';
+import ExternalRequestLayout from '@popup/page-layouts/ExternalRequestLayout';
+import { fullscreenPromptContext } from '@popup/page-layouts/FullscreenPromptLayout';
+import Button from '@popup/shared/Button';
+import ButtonGroup from '@popup/shared/ButtonGroup';
+import { displayUrl } from '@popup/shared/utils/string-helpers';
+import {
+    storedVerifiableCredentialsAtom,
+    storedVerifiableCredentialSchemasAtom,
+} from '@popup/store/verifiable-credential';
+import { useConfirmedIdentities } from '@popup/shared/utils/identity-helpers';
+import { parse } from '@shared/utils/payload-helpers';
+import { DisplayCredentialStatement } from './DisplayStatement';
+import { getCommitmentInput } from './utils';
+
+type Props = {
+    onSubmit(presentationString: string): void;
+    onReject(): void;
+};
+
+interface Location {
+    state: {
+        payload: {
+            challenge: string;
+            statements: string;
+            url: string;
+        };
+    };
+}
+
+export default function Web3ProofRequest({ onReject, onSubmit }: Props) {
+    const { state } = useLocation() as Location;
+    const { statements: rawStatements, challenge, url } = state.payload;
+    const { onClose, withClose } = useContext(fullscreenPromptContext);
+    const { t } = useTranslation('web3IdProofRequest');
+    const network = useAtomValue(networkConfigurationAtom);
+    const client = useAtomValue(grpcClientAtom);
+    const addToast = useSetAtom(addToastAtom);
+    const recoveryPhrase = useDecryptedSeedPhrase((e) => addToast(e.message));
+    const dappName = displayUrl(url);
+    const [creatingProof, setCreatingProof] = useState<boolean>(false);
+    const [currentStatementIndex, setCurrentStatementIndex] = useState<number>(0);
+    const net = getNet(network);
+
+    const statements: CredentialStatements = useMemo(() => parse(rawStatements), [rawStatements]);
+
+    const [ids, setIds] = useState<string[]>(Array(statements.length).fill(''));
+
+    const verifiableCredentialSchemas = useAtomValue(storedVerifiableCredentialSchemasAtom);
+    const identities = useConfirmedIdentities();
+    const credentials = useAtomValue(credentialsAtom);
+    const verifiableCredentials = useAtomValue(storedVerifiableCredentialsAtom);
+
+    const canProve = useMemo(() => ids.every((x) => Boolean(x)), [ids]);
+
+    const handleSubmit = useCallback(async () => {
+        if (!recoveryPhrase) {
+            throw new Error('Missing recovery phrase');
+        }
+        if (!network) {
+            throw new Error('Network is not specified');
+        }
+        if (!ids.every((x) => Boolean(x))) {
```

Comment:
This is a duplicate of `canProve` as far as I can see. Also, the message in the `Error` does not match what the check is for.